pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -220,6 +220,14 @@ func (c *ClusterMemberController) getEtcdPodToAddToMembership() (*corev1.Pod, er
 			continue
 		}
 
+		if (pod.Status.InitContainerStatuses[1].State.Terminated == nil) ||
```

Comment:
The intention here is to make sure that we have a:
1. Terminated state on the init containers. This will make sure that they are not in Running state
2. The exit code on the Terminated state is 0.

Reply:
Perhaps add this comment in the code.

Reply:
Just realised maybe I can do simpler. Both the conditions are satisfied if "etcd" container is running, I will try to replace the if condition